sunpy/map/compositemap.py

```diff
@@ -444,7 +444,10 @@ def plot(self, axes=None, annotate=True,
             if m.levels is False:
                 # We tell GenericMap.plot() that we need to autoalign the map
                 if wcsaxes_compat.is_wcsaxes(axes):
-                    params['autoalign'] = True
+                    if m.wcs == axes.wcs:
```

Comment:
You can't simply check for equality this way.  WCSs can be the same without being strictly equal.  I believe we have code that can be copied from `mapbase.py`.

Reply:
Hey @ayshih 
there is method in https://github.com/astropy/astropy/blob/53b673bb4fd98da7b8972837a9650469bc130b02/cextern/wcslib/C/wcs.c#L1478-L1480
that implement the same thing

I think it is better to use that 

```
import sunpy.data.sample
import sunpy.map

aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
hmi_map = sunpy.map.Map(sunpy.data.sample.HMI_LOS_IMAGE)
print(aia_map.wcs.wcs.compare(hmi_map.wcs.wcs))
```
it is also dealing with some tolerance value

Reply:
we can see that   
 In function `GenericMap.reproject_to` https://github.com/sunpy/sunpy/blob/499a03cc58f0556d44658fb7ff18c830146cfdc1/sunpy/map/mapbase.py#L3007 
returns  GenericMap containing only `target wcs.header`  info

should i proceed to check only with that values

```
import sunpy.data.sample
import sunpy.map
map_aia = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
map_hmi = sunpy.map.Map(sunpy.data.sample.HMI_LOS_IMAGE)

out_hmi = map_hmi.reproject_to(map_aia.wcs)
print(out_hmi.wcs.to_header() == map_aia.wcs.to_header())  // True 
print(out_hmi.wcs == map_aia.wcs)  // False

```

Reply:
I have changed the equality logic for wcs, and tests according to the need.
should i include `datebeg, dateend` ?
https://github.com/sunpy/sunpy/blob/499a03cc58f0556d44658fb7ff18c830146cfdc1/sunpy/map/mapbase.py#L638-L641